src/main/clojure/clara/tools/testing_utils.clj

```diff
@@ -0,0 +1,66 @@
+  clara.test-testing-utils.  Note that sessions currently can only contain rules and queries defined
+  in the setup map; supporting other rule sources such as namespaces and defrule/defquery may be possible
+  in the future.
+
+  Namespaces consuming this macro are expected to require clara.rules and either clojure.test or cljs.test.
+  Unfortunately, at this time we can't add inline requires for these namespace with the macroexpanded code in
+  ClojureScript; see https://anmonteiro.com/2016/10/clojurescript-require-outside-ns/ for some discussion on the 
+  subject.  However, the test namespaces consuming this will in all likelihood have these dependencies anyway
+  so this probably isn't a significant shortcoming of this macro."
+  [name params & forms]
+  (let [sym->rule (->> params
+                       :rules
+                       (partition 2)
+                       (into {}
+                             (map (fn [[rule-name [lhs rhs props]]]
+                                    [rule-name (dsl/parse-rule* lhs rhs props {})]))))
+
+        sym->query (->> params
+                        :queries
+                        (partition 2)
+                        (into {}
+                              (map (fn [[query-name [params lhs]]]
+                                     [query-name (dsl/parse-query* params lhs {})]))))
+        
+        production-syms->productions (fn [p-syms]
+                                       (map (fn [s]
+                                              (or (get sym->rule s)
+                                                  (get sym->query s)))
+                                            p-syms))
+        
+        session-syms->session-forms (->> params
+                                         :sessions
+                                         (partition 3)
+                                         (into []
+                                               (comp (map (fn [[session-name production-syms session-opts]]
+                                                            [session-name (production-syms->productions production-syms) session-opts]))
+                                                     (map (fn [[session-name productions session-opts]]
+                                                            [session-name (if (com/compiling-cljs?)
+                                                                            (m/productions->session-assembly-form (map eval productions) session-opts)
+                                                                            `(com/mk-session ~(into [(vec productions)]
+                                                                                                    cat
+                                                                                                    session-opts)))]))
+                                                     cat)))
+        
+        test-form `(~(if (com/compiling-cljs?)
+                       'cljs.test/deftest
```

Comment:
This is sort of a macro-anti-pattern to return code that has implicit dependencies on namespaces being pre-required by the caller.

If this macro was used by a caller that did not have a `require` on `cljs.test` (or clj) and it wasn't already loaded elsewhere, then it would fail.